odoo/addons/base/tests/test_translate.py

```diff
@@ -874,6 +874,57 @@ def test_sync_xml(self):
         self.assertEqual(view.with_env(env_fr).arch_db, archf % (terms_en[0], terms_fr[1]))
         self.assertEqual(view.with_env(env_nl).arch_db, archf % (terms_en[0], terms_nl[1]))
 
+    def test_sync_xml_collision(self):
+        """ Check translations of 'arch' after xml tags changes in source terms
+            when the same term appears in different elements with different
+            styles.
+        """
+        archf = '''<form class="row s_nb_column_fixed">
+    %s
+    <div class="col-lg-3 s_table_of_content_navbar_wrap s_table_of_content_navbar_sticky s_table_of_content_vertical_navbar d-print-none d-none d-lg-block o_not_editable o_cc o_cc1 o_colored_level" data-name="Navbar" contenteditable="false">
+        <div class="s_table_of_content_navbar list-group o_no_link_popover" style="top: 76px;"><a href="#table_of_content_heading_1672668075678_4" class="table_of_content_link list-group-item list-group-item-action py-2 border-0 rounded-0 active">%s</a></div>
+    </div>
+    <div class="col-lg-9 s_table_of_content_main oe_structure oe_empty o_colored_level" data-name="Content">
+        <section class="pb16 o_colored_level" style="background-image: none;">
+            <h1 data-anchor="true" class="o_default_snippet_text" id="table_of_content_heading_1672668075678_4">%s</h1>
+        </section>
+    </div>
+</form>'''
```

Comment:
May I suggest some simplification of the HTML?  The list of classes and style has no relevance in this test and only introduce a lot of noise.  I struggled to find the `%s` into that stuff
```suggestion
        archf = '''<form class="row">
    %s
    <div class="s_table_of_content_vertical_navbar" data-name="Navbar" contenteditable="false">
        <div class="s_table_of_content_navbar"><a href="#table_of_content_heading_1672668075678_4" class="table_of_content_link">%s</a></div>
    </div>
    <div class="s_table_of_content_main" data-name="Content">
        <section class="pb16">
            <h1 data-anchor="true" class="o_default_snippet_text" id="table_of_content_heading_1672668075678_4">%s</h1>
        </section>
    </div>
</form>'''
```